### PR TITLE
Group containers by compose project name

### PR DESCRIPTION
**Pull request by ravipal**
A new property "Compose Project Name" is added to container. This will allow group by compose projects. This will also enable user to choose this new propery to be displayed on tree item label and description.
Fixes: #1846

### CODE REVIEW

File: src/tree/containers/ContainerProperties.ts [hunk 1]
```diff
@@ -16,7 +16,8 @@ export const containerProperties: ITreePropertyInfo<ContainerProperty>[] = [
     { property: 'Networks', exampleValue: 'mybridge_network' },
     { property: 'Ports', exampleValue: '8080' },
     { property: 'State', exampleValue: 'exited' },
-    { property: 'Status', exampleValue: 'Exited (0) 2 hours ago' }
+    { property: 'Status', exampleValue: 'Exited (0) 2 hours ago' },
+    { property: 'Compose Project Name', description: 'Value used to associate containers launched by a \'compose up\' command' },
```
Comment (karolz-ms) [inline src/tree/containers/ContainerProperties.ts:20]: Should description be localizable?

File: src/tree/containers/ContainersTreeItem.ts [hunk 2]
```diff
@@ -67,8 +67,26 @@ export class ContainersTreeItem extends LocalRootTreeItemBase<ILocalContainerInf
...
             default:
                 return getImagePropertyValue(item, property);
         }
     }
+
+    public compareChildrenImpl(ti1: ContainerTreeItem, ti2: ContainerTreeItem): number {
+        if (this.failedToConnect) {
+            return 0; // children are already sorted
+        }
+        if (this.groupBySetting === 'Compose Project Name'
```
Comment (karolz-ms) [inline src/tree/containers/ContainersTreeItem.ts:81]: I assume this is to ensure that non-Compose group is at the bottom. Consider adding a comment clarifying this intent.

File: src/tree/containers/LocalContainerInfo.ts [hunk 3]
```diff
@@ -18,8 +19,10 @@ export interface ILocalContainerInfo extends ILocalItem {
     ports: number[];
     state: string;
     status: string;
+    compose: string;
 }
 
+export const NonComposeGroupName = 'Other Containers';
```
Comment (karolz-ms) [inline src/tree/containers/LocalContainerInfo.ts:25]: Localizable?

File: src/tree/containers/LocalContainerInfo.ts [hunk 4]
```diff
@@ -75,6 +78,18 @@ export class LocalContainerInfo implements ILocalContainerInfo {
         return this.data.Status;
     }
 
+    public get compose(): string {
```
Comment (karolz-ms) [inline src/tree/containers/LocalContainerInfo.ts:81]: `composeProjectName`?

File: package.json [hunk 5]
```diff
@@ -1599,6 +1600,7 @@
                     "items": {
                         "type": "string",
                         "enum": [
+                            "Compose Project Name",
```
Comment (bwateratmsft) [inline package.json:1603]: Many of the other settings that are multiple words don't have spaces, do we know why that is? e.g. `RepositoryNameAndTag`

File: src/tree/containers/LocalContainerInfo.ts [hunk 6]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/tslint/config */
```
Comment (bwateratmsft) [inline src/tree/containers/LocalContainerInfo.ts:1]: Please disable only the lines where it is needed, not the whole file.

File: package.json (hunk 5 above)
Comment (ravipal) [reply package.json:1603]: We had a chat about this. Most of the property names are coming from docker and hence with no space. "Compose Project Name" is not exactly a docker property, we derive this value from a label.

Comment (bwateratmsft) [reply package.json:1603]: Gotcha, makes sense.

File: src/tree/containers/ContainerProperties.ts [hunk 7]
```diff
@@ -16,7 +17,8 @@ export const containerProperties: ITreePropertyInfo<ContainerProperty>[] = [
     { property: 'Networks', exampleValue: 'mybridge_network' },
     { property: 'Ports', exampleValue: '8080' },
     { property: 'State', exampleValue: 'exited' },
-    { property: 'Status', exampleValue: 'Exited (0) 2 hours ago' }
+    { property: 'Status', exampleValue: 'Exited (0) 2 hours ago' },
+    { property: 'Compose Project Name', description: localize('vscode-docker.tree.containers.properties.composeProjectName', 'Value used to associate containers launched by a \'compose up\' command') },
```
Comment (bwateratmsft) [inline src/tree/containers/ContainerProperties.ts:21]:
```suggestion
    { property: 'Compose Project Name', description: localize('vscode-docker.tree.containers.properties.composeProjectName', 'Value used to associate containers launched by a \'docker-compose up\' command') },
```
I guess we should probably use the full name here.

Comment (ravipal) [reply src/tree/containers/ContainerProperties.ts:21]: It is already a long description and user will get 'compose up'. @BigMorty ?

**Pull request merged**